Automattic/wp-calypso
# ChartTabs: Optimize performance using the data layer

This focuses on improving performance and maintainability of the Stats Dashboard (`/stats/day`, `/stats/week`, `/stats/month`, `/stats/year`). Namely, it achieves the following:

- Reducing redundancy in the two API requests used to populate the data.
  - Previously, we made a request for the current chart tab and another request for all tabs.
  - Now, we make a request for the current chart tab and another for the unselected tabs.
- Implements a Redux state using Calypso's data layer (+ schemas and tests).
- Reduces the amount of computation done on every render.

#### Testing instructions

1. Check out this branch locally and run the client tests; ensure they pass.
```shell
npm run test-client
```
2. Spin this branch up locally or via calypso.live.
3. Navigate to [`/stats/day`](http://calypso.localhost:3000/stats/day) and select a site. 
4. Ensure that the graph loads as expected. Also, verify that only two API requests to `/stats/visits` are made in the network console in your browser.
